scripts/dev/generate_config_types.py

```diff
@@ -0,0 +1,236 @@
+"""This script auto-generates the `qutebrowser/config/configcontainer.py` file."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from textwrap import dedent
+from typing import Union, cast
+from collections.abc import Mapping
+from typing_extensions import TypeAlias, assert_never
```

Comment:
`typing_extensions` would be a new dev dep but I couldn't figure out how to add it? After updating `misc/requirements/requirements-dev.txt-raw` and running `python scripts/dev/recompile_requirements.py` I got a bunch of diffs

Reply:
> even more so since TypeAlias is deprecated in 3.12 again

oh 🤦 I had no idea lol